### PR TITLE
Do not break analysing other picture types

### PR DESCRIPTION
**Pull request by tvdeyen**
First of all thanks for this great plugin!

Unfortunately with this plugin enabled the build in `:image_magick` analyser is broken,
because this plugin overrides existing analysers (`:width`, `:height`, etc.).
After this, it checks the image type and analysises accordingly.

Also removes redundant `:landscape` analyser.
Dragonfly already ships with a `:landscape` analyser that has the same exact implemenation.

### CODE REVIEW

**Comment by tomasc** [comment]
Hi @tvdeyen , thank for the PR.

The master is updated to make the Travis tests pass, please rebase.

I would like to avoid calling `content.analyse(:image_properties)` explicitly, which may or may not be plugged in. Did you try to return `false` from the analyser, use `super` or something along these lines in order to hand over to the next one?

**Comment by tvdeyen** [comment]
Yes. This was my first approach. I also don't like the current implementation. 

Unfortunately the interface of dragonfly does not provide this. There is no super class, because these analyzers are local blocks stored in a hash and your plugin overrides the original analyzers defined by the image magic plugin. So, we only have this one analyzer available and need to reimplement the logic of the image magick plugin. This would need some serious refactoring in dragonfly. 

But you're right, we should check if the image magick analyzer is available, before using it. And we could refactor your plugin do avoid as much duplication as possible. 

Maybe I can invest some time this evening. 

Please do chip in, if you have something in mind. 

Thanks

PS: As I'm writing this, maybe we could store the original analyzes before overwriting them. Yes, this sounds legit.

**Comment by tomasc** [comment]
Aww, okay.
Please go ahead and give it a try – checking if there's something already defined, storing it, and calling back if the format is not supported sounds reasonable.
Perhaps it might be a good idea to file an issue on dragonfly, as I guess it'd make sense to have a standard solution for conflicting analysers upstream in the end.

**Comment by tvdeyen** [comment]
Yes. Will do.

It would be great to register a dragonfly plugin for certain types of files only. So, that dragonfly calls all analyzes available for a certain kind of file. But this is way out the scope of my pull request.

**Comment by tomasc** [comment]
Yes, good idea. Should be handled in dragonfly itself for sure.

**Comment by tvdeyen** [comment]
@tomasc done.

File: lib/dragonfly_svg/analysers/aspect_ratio_analyser.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+      def initialize(app)
+        original = app.analysers.items[:aspect_ratio]
+        app.add_analyser(:aspect_ratio) do |content|
+          if content.mime_type == Plugin::SVG_MIME_TYPE
+            attrs = content.analyse(:svg_properties)
+            attrs[:width].to_f / attrs[:height]
+          elsif original
+            original.call(content)
+          end
```
Comment (tomasc) [inline lib/dragonfly_svg/analysers/aspect_ratio_analyser.rb:13]: Would it be cleaner to replace this with:

``` ruby
return original.call(content) unless content.mime_type == Plugin::SVG_MIME_TYPE
attrs = content.analyse(:svg_properties)
attrs[:width].to_f / attrs[:height]
```

File: lib/dragonfly_svg/analysers/height_analyser.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+      def initialize(app)
+        original = app.analysers.items[:height]
+        app.add_analyser(:height) do |content|
+          if content.mime_type == Plugin::SVG_MIME_TYPE
+            content.analyse(:svg_properties)[:height]
+          elsif original
+            original.call(content)
+          end
+        end
```
Comment (tomasc) [inline lib/dragonfly_svg/analysers/height_analyser.rb:13]: same as above

File: lib/dragonfly_svg/analysers/landscape_analyser.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+      def initialize(app)
+        original = app.analysers.items[:landscape]
+        app.add_analyser(:landscape) do |content|
+          if content.mime_type == Plugin::SVG_MIME_TYPE
+            attrs = content.analyse(:svg_properties)
+            attrs[:width] >= attrs[:height]
+          elsif original
+            original.call(content)
+          end
```
Comment (tomasc) [inline lib/dragonfly_svg/analysers/landscape_analyser.rb:13]: same as above

File: lib/dragonfly_svg/analysers/portrait_analyser.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+      def initialize(app)
+        original = app.analysers.items[:portrait]
+        app.add_analyser(:portrait) do |content|
+          if content.mime_type == Plugin::SVG_MIME_TYPE
+            attrs = content.analyse(:svg_properties)
+            attrs[:width] <= attrs[:height]
+          elsif original
+            original.call(content)
+          end
```
Comment (tomasc) [inline lib/dragonfly_svg/analysers/portrait_analyser.rb:13]: same as above

File: lib/dragonfly_svg/analysers/width_analyser.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+      def initialize(app)
+        original = app.analysers.items[:width]
+        app.add_analyser(:width) do |content|
+          if content.mime_type == Plugin::SVG_MIME_TYPE
+            content.analyse(:svg_properties)[:width]
+          elsif original
+            original.call(content)
+          end
+        end
```
Comment (tomasc) [inline lib/dragonfly_svg/analysers/width_analyser.rb:13]: same as above

**Comment by tomasc** [comment]
great, thank you!
I have a small suggestion regarding the `original.call` syntax. Please take a look if that makes sense?

File: lib/dragonfly_svg/analysers/aspect_ratio_analyser.rb (hunk 1 above)
Comment (tvdeyen) [inline lib/dragonfly_svg/analysers/aspect_ratio_analyser.rb:13]: We would need to check if `original` is present before calling it. A simple `original.try(:call, content)` should do the trick. Will update the code.

**Comment by tvdeyen** [comment]
@tomasc done

**Comment by tomasc** [comment]
hmm, tests are failing – `undefined local variable or method 'content'`

**Comment by tvdeyen** [comment]
my bad. fixed

**Comment by tvdeyen** [comment]
Aw come on. To much `ActiveSupport` makes bad ruby devs. Sorry :)

**Comment by tvdeyen** [comment]
@tomasc the thing about the failing specs:

We really need to check if `original` is present and only call it then. So, I don't think the refactoring is worth the hassle here. `try` is not part of POR, and requiring `active_support` isn't an option. So, I would say: ditch the refactoring, my first version is good enough. You can always optimize later

**Comment by tomasc** [comment]
@tvdeyen yup, agreed, let's revert

**Comment by tvdeyen** [comment]
One last try ;)

**Comment by tvdeyen** [comment]
Ok, reverted @tomasc

**Comment by tvdeyen** [comment]
Phew finally 😀

**Comment by tomasc** [comment]
🔨

**Comment by tvdeyen** [comment]
@tomasc Thanks! Would you mind to release a new gem version soon, so I can add it as gem dependency in another gem? That would be awesome 👍

**Comment by tomasc** [comment]
yep, i'll merge the setAttribute and release

**Comment by tvdeyen** [comment]
Great! Thanks

**Comment by tvdeyen** [comment]
BTW: You're great plugin will be used as default SVG analyzer and processor in https://github.com/AlchemyCMS/alchemy_cms soon.

Thanks for your work.

**Comment by tomasc** [comment]
great, nice to hear, `0.0.4` gem's out

**Pull request merged**